### ISSUE TITLE
Error Response is not available for WebhookCallFailedEvent

### ISSUE THREAD

**Issue by 2c-borispopov** [spatie/laravel-webhook-server#20]
When `WebhookCallFailedEvent`(or `FinalWebhookCallFailedEvent`) events are dispatched, there is no response property in the event objects.

This is because when Guzzle exception is thrown `$this->response` is not being set in the handle method.

A trivial fix would be to add 

```php
$this->response = $exception->getResponse();
```

in the catch block of `CallWebhookJob::handle` method:

```php
    public function handle()
    {
        /** @var \GuzzleHttp\Client $client */
        $client = app(Client::class);

        try {
            $this->response = $client->request($this->httpVerb, $this->webhookUrl, [
                'timeout' => $this->requestTimeout,
                'body' => json_encode($this->payload),
                'verify' => $this->verifySsl,
                'headers' => $this->headers,
            ]);

            if (! Str::startsWith($this->response->getStatusCode(), 2)) {
                throw new Exception('Webhook call failed');
            }

            $this->dispatchEvent(WebhookCallSucceededEvent::class);

            return;
        } catch (Exception $exception) {
            $this->response = $exception->getResponse();

            /** @var \Spatie\WebhookServer\BackoffStrategy\BackoffStrategy $backoffStrategy */
            $backoffStrategy = app($this->backoffStrategyClass);

            $waitInSeconds = $backoffStrategy->waitInSecondsAfterAttempt($this->attempts());

            $this->dispatchEvent(WebhookCallFailedEvent::class);

            $this->release($waitInSeconds);
        }

        if ($this->attempts() >= $this->tries) {
            $this->dispatchEvent(FinalWebhookCallFailedEvent::class);

            $this->delete();
        }
    }
```

**Comment by dmason30** [comment]
@2c-borispopov This was merged and added in 1.20 today I had previously raised this and created a PR.

**Comment by pakcybershagufta** [comment]
still no fix??
